src/Native/Web3.js

```diff
@@ -58,11 +63,28 @@ var _cmditch$elm_web3$Native_Web3 = function() {
                     f.apply(null, request.args.concat( web3Callback ));
                 } else if (request.callType.ctor === "Sync") {
                     var syncResult = f.apply(null, request.args);
-                    syncResult = request.expect.responseToResult( formatWeb3Response(syncResult) );
-                    return callback(_elm_lang$core$Native_Scheduler.succeed(syncResult._0));
+                /* */
+                /* */
+                    // web3.reset() returns undefined and needs to be handled accordingly
+                    if (syncResult === undefined && request.func === "reset") {
```

Comment:
This conditional jankery was needed for one web3 function which returns `undefined` regardless.

We might be able to significantly clean up the core Web3.js file. But would like to get full coverage and tests in place first. Then focus on A) Optimization, B) Cleanliness